[GFC][Cleanup] Move type aliases to dedicated header.

sammygill opened the pull request:
#### 60e481756263447736aae90d752f5312ae6e3746
<pre>
[GFC][Cleanup] Move type aliases to dedicated header.
<a href="https://bugs.webkit.org/show_bug.cgi?id=299615">https://bugs.webkit.org/show_bug.cgi?id=299615</a>
<a href="https://rdar.apple.com/problem/161422019">rdar://problem/161422019</a>

Reviewed by Brent Fulgham.

We have various type alises that we use throughout different files in
GFC. Sometimes this results in a bunch of redundant declarations
when the type alias or some other forward declaration is not included.
If we move these type alises along with the forward declarations needed
for them into a dedicated header file this significantly reduces the
when reading the rest of the code.

Along with the change above there are also a couple of other related
changes:
1.) Move the UsedTrackSizes and UnsizedTrack struct definitions from
the headers to the their respective sources since they are currently
not used outside of thier translation units.

2.) Move TrackSizingFunctions to its own header.

Canonical link: <a href="https://commits.webkit.org/300604@main">https://commits.webkit.org/300604@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/226800149d9585ff3c7fcf131f51fbce71a2569e

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/123182 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/42896 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/33593 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/129889 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/75294 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/e2551b84-ac8f-4d19-90db-f2b6b0491cfa) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/125059 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/43620 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/51491 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/93639 "Found 1 new test failure: imported/w3c/web-platform-tests/html/semantics/forms/constraints/infinite_backtracking.html (failure)") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/62131 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/36d1f53a-0297-4bd6-b7a4-2fc34f42f534) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/126133 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/34764 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/110243 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/74269 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/b5f6c993-8873-4af7-8e7e-b417290786b4) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/33732 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/28399 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/73405 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/104476 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/28625 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/132604 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/50132 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/38165 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/102131 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/50508 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/106464 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/101987 "Found 1 new API test failure: WebKitGTK/TestInspector:/webkit/WebKitWebInspector/manual-attach-detach (failure)") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/25931 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/47359 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/25567 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/46906 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/49987 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/55748 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/49457 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/52807 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/51136 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

brentfulgham commented on Source/WebCore/layout/formattingContexts/grid/GridTypeAliases.h:
```diff
@@ -0,0 +1,49 @@
+ */
+
+#pragma once
+
```
We might want to:

`#include <wtf/Forward.h>`

... to get all the forward declarations for Vector/HashMap, etc. Not a big deal if this file is always used in implementation files that already have that stuff, but could avoid build breakage if you start using this header in more places.

brentfulgham commented on Source/WebCore/layout/formattingContexts/grid/TrackSizingAlgorithm.cpp:
```diff
@@ -31,8 +31,14 @@
+struct UnsizedTrack {
+    LayoutUnit baseSize;
+    LayoutUnit growthLimit;
+    const TrackSizingFunctions trackSizingFunction;
```
I'm surprised this struct and `UsedTrackSizes` can be forward declared and only defined in these implementation files! I wonder if we should (eventually) make another `*Aliases.h` header at some point for these types. Surely someone in another file will eventually want access a member of this structure? (But perhaps not...)

brentfulgham replied:
I guess these were private definitions, so maybe it's totally fine for them to be in the implementation file after all.

sammygill replied:
Yup but we can totally move it to the header if we need to do it at some point! Just did it to keep it smaller for now

brentfulgham reviewed (approved):
Looks great! Thank you for this cleanup. Looks much tidier to me :-)

sammygill committed:
[GFC][Cleanup] Move type aliases to dedicated header.
https://bugs.webkit.org/show_bug.cgi?id=299615
rdar://problem/161422019

Reviewed by Brent Fulgham.

We have various type alises that we use throughout different files in
GFC. Sometimes this results in a bunch of redundant declarations
when the type alias or some other forward declaration is not included.
If we move these type alises along with the forward declarations needed
for them into a dedicated header file this significantly reduces the
when reading the rest of the code.

Along with the change above there are also a couple of other related
changes:
1.) Move the UsedTrackSizes and UnsizedTrack struct definitions from
the headers to the their respective sources since they are currently
not used outside of thier translation units.

2.) Move TrackSizingFunctions to its own header.

Canonical link: https://commits.webkit.org/300604@main
